src/lib/moderation.ts

```diff
@@ -0,0 +1,30 @@
+import { json } from '@sveltejs/kit';
+import tf, { type Tensor3D } from '@tensorflow/tfjs-node';
+import nsfw from 'nsfwjs';
+
+const BAD_PREDICTION_TYPES = ['Hentai', 'Porn'];
+const PREDICTION_TRESHOLD = 0.7;
+export const NSFW_ERROR = json(
+	{
+		error:
+			'It seems like you just uploaded an NSFW image. We do not allow this type of content, if you believe this is a mistake, please contact us.'
+	},
+	{ status: 400 }
+);
+
+tf.enableProdMode();
+const model = await nsfw.load();
```

Comment:
You should make the model local `await nsfw.load(`file://.
/models/mobilenet_v2_mid/`, { type: 'graph' })`

You can get the files here:
https://github.com/infinitered/nsfwjs/tree/master/models/mobilenet_v2_mid

Reply:
why?

Reply:
> why?

```The magic that powers NSFWJS is the NSFW detection model. By default, this node module is pulling from my S3, but I make no guarantees that I'll keep that download link available forever. It's best for the longevity of your project that you download and host your own version of the model files. You can then pass the relative URL to your hosted files in the load function. If you can come up with a way to bundle the model into the NPM package, I'd love to see a PR to this repo!```
That's what it says on the nsfwjs npm page